api/v1alpha1/ionoscloudmachine_types.go

```diff
@@ -240,13 +240,32 @@ type ImageSpec struct {
 	Selector *ImageSelector `json:"selector,omitempty"`
 }
 
+// ImageSelectorResolutionPolicy defines what action to take when the image selector has ambiguous results.
+// +kubebuilder:validation:Enum=Exact;Newest
+type ImageSelectorResolutionPolicy string
+
+const (
+	// ResolutionPolicyExact only succeeds if the image selector resolves to exactly 1 image.
+	ResolutionPolicyExact ImageSelectorResolutionPolicy = "Exact"
+	// ResolutionPolicyNewest uses the newest entry if the image selector resolves to more than 1 image.
+	ResolutionPolicyNewest ImageSelectorResolutionPolicy = "Newest"
+)
+
 // ImageSelector defines label selectors for looking up images.
 type ImageSelector struct {
 	// MatchLabels is a map of key/value pairs.
 	//
 	//+kubebuilder:validation:MinProperties=1
 	MatchLabels map[string]string `json:"matchLabels"`
 
+	// ResolutionPolicy controls the lookup behavior.
+	// The default policy 'Exact' will raise an error if the selector resolves to more than 1 image.
+	// Use policy 'Newest' to select the newest image instead.
+	//
+	//+kubebuilder:default=`Exact`
```

Comment:
if your manifest doesn't include `resolutionPolicy,` the API will always set it to `Exact.`

Reply:
Perhaps the applied manifest is without the `resolutionPolicy` but the current state would be with `Exact`
Is that okay? 
Otherwise, we have the option to use a pointer to determine that a nil value is also `Exact`

Reply:
i wanted the policy to be explicit. if we update the meaning of "not set" it's a behavior change for users that never bothered to set it, whereas having this as value and always present keeps the behavior stable.